interfaces/cython/cantera/thermo.pyx

```diff
@@ -268,7 +268,11 @@ cdef class ThermoPhase(_SolutionBase):
             self.thermo_basis = mass_basis
         # In composite objects, the ThermoPhase constructor needs to be called first
         # to prevent instantiation of stand-alone 'Kinetics' or 'Transport' objects.
-        # The following is used as a sentinel.
+        # The following is used as a sentinel. After initialization, the _references
```

Comment:
I know this is still in draft mode, but aren’t sentinels for these cases relevant to other API’s (not just Python) as well? In other words, could these be set in C++?

Reply:
Yes, this feature probably ought to be implemented at the C++ level, but doing so is somewhat complicated. I think some of the things you've done recently to more generally use `shared_ptr` may make this easier after Cantera 3.0.

Reply:
I'm :+1: with keeping things Python-specific for the moment, but would request a note/comment ...